### PR TITLE
Update minimum required versions of some dependencies

### PR DESCRIPTION
**Pull request by vtnate**
#### Any background context you want to provide?
A new version of Ruff [was released](https://astral.sh/blog/ruff-v0.6.0) that I wanted to take advantage of. Doing so inspired several attached improvements:
- Update pytest version from 7 -> 8
- Use carets `^` in a few more dev dependencies to be a little more permissive (according to the [poetry docs](https://python-poetry.org/docs/dependency-specification/))
- Move Ruff config to its own dedicated file
- Address the linting items raised by this new Ruff version
#### What does this PR accomplish?
- Use newer versions of several dependencies
- Use newer versions of pre-commit hooks, including Ruff
- Changes from Ruff
- Ruff config is now in a dedicated file. The intent is to simplify/clean up pyproject.toml.
#### How should this be manually tested?
CI is sufficient technically. Review diff of `pyproject.toml` to ensure the dependency version changes are desired.

### CODE REVIEW

File: geojson_modelica_translator/geojson/schemas.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(self):
         }
 
         for s in self.schemas:
-            path = os.path.join(os.path.dirname(__file__), "data/schemas/%s_properties.json" % s)
+            path = Path(__file__).parent / "data" / "schemas" / f"{s}_properties.json"
```
Comment (nllong) [inline geojson_modelica_translator/geojson/schemas.py:27]: haha, you continue to clean up the use of `os.path`.

File: tests/system_parameters/test_system_parameters.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def test_valid_system_parameters_ghe(self):
         sdp = SystemParameters(filename)
         assert sdp is not None
         assert len(sdp.validate()) == 0
-        assert [] == sdp.validate()
+        assert sdp.validate() == []
```
Comment (nllong) [inline tests/system_parameters/test_system_parameters.py:70]: strange update... I assume ruff did this.

**Review by nllong — APPROVED** [review]
sweet.

File: geojson_modelica_translator/geojson/schemas.py (hunk 1 above)
Comment (vtnate) [reply geojson_modelica_translator/geojson/schemas.py:27]: 🤓

Comment (vtnate) [reply geojson_modelica_translator/geojson/schemas.py:27]: First Ruff didn't like the use of `%s`, so I was touching that line anyway. And yeah, my neuroses make me use Pathlib 😆

**Pull request merged**